File: server/datasets/python_source-code/Multiple-B1/exact_copy_recursive_1.py
# Binary Tree Traversal - unique source, RECURSIVE family
# This is one of two "true source" files for this test case. It should
# match exact_copy_recursive_1.py and (partially, via disguise) match
# renamed_reordered_combo_1.py - but must NOT cross-match with the
# iterative family (unique_iterative_1.py / exact_copy_iterative_1.py).

class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None


def insert(root, value):
    if root is None:
        return Node(value)
    if value < root.value:
        root.left = insert(root.left, value)
    else:
        root.right = insert(root.right, value)
    return root


def inorder_recursive(node, result):
    if node is None:
        return
    inorder_recursive(node.left, result)
    result.append(node.value)
    inorder_recursive(node.right, result)


def build_tree(values):
    root = None
    for v in values:
        root = insert(root, v)
    return root


if __name__ == "__main__":
    values = [50, 30, 70, 20, 40, 60, 80]
    root = build_tree(values)
    result = []
    inorder_recursive(root, result)
    print("In-order traversal (recursive):", result)

File: server/datasets/java_source-code/Multiple-B1/exact_copy_recursive_1.java
// Binary Tree Traversal - unique source, RECURSIVE family
// This is one of two "true source" files for this test case.

import java.util.ArrayList;
import java.util.List;

class Node {
    int value;
    Node left;
    Node right;

    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

class Solution {
    public static Node insert(Node root, int value) {
        if (root == null) {
            return new Node(value);
        }
        if (value < root.value) {
            root.left = insert(root.left, value);
        } else {
            root.right = insert(root.right, value);
        }
        return root;
    }

    public static void inorderRecursive(Node node, List<Integer> result) {
        if (node == null) {
            return;
        }
        inorderRecursive(node.left, result);
        result.add(node.value);
        inorderRecursive(node.right, result);
    }

    public static Node buildTree(int[] values) {
        Node root = null;
        for (int v : values) {
            root = insert(root, v);
        }
        return root;
    }

    public static void main(String[] args) {
        int[] values = {50, 30, 70, 20, 40, 60, 80};
        Node root = buildTree(values);
        List<Integer> result = new ArrayList<>();
        inorderRecursive(root, result);
        System.out.println("In-order traversal (recursive): " + result);
    }
}
